src/test/java/echoServer/AppTest.java

```diff
@@ -4,11 +4,50 @@
 package echoServer;
 
 import org.junit.Test;
+import org.junit.Before;
+import org.junit.After;
 import static org.junit.Assert.*;
 
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.io.PrintStream;
+
 public class AppTest {
+    App classUnderTest;
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+    private final InputStream originalIn = System.in;
+    private final PrintStream originalOut = System.out;
+
+    @Before public void initialize() {
+        classUnderTest = new App();
+    }
+
+    @Before public void setupStreams() {
+        System.setOut(new PrintStream(outContent));
+    }
```

Comment:
Is there a particular reason that you chose to initialize the `App` and set up the streams in two separate `@Before`s?

Reply:
No particular reason. It didn't cross my mind that I could declare the `setupStreams()` method outside of the `@Before` tag and simply initialize the `App` and call `setupStreams()` within one `@Before` tag.